You,have,a,situation,that,looks,like,this,:,i.e.,three,references,to,the,same,underlying,object,.,All,that,del,h,does,is,remove,the,name,h,",",like,this,:,The,underlying,object,doesn't,change,",",and,the,two,other,names,still,provide,a,reference,to,it,",",all,you've,done,is,remove,one,reference,to,the,Foo,instance,.,h,is,now,a,NameError,",",but,g,and,f.attr,will,work,just,fine,.,I,don't,know,why,you,would,conclude,that,-,if,you,did,succeed,in,breaking,the,reference,from,f.attr,to,the,Foo,instance,",",you,would,get,an,AttributeError,",",not,None,.,Recommended,reading,on,Python,names,:,http://nedbatchelder.com/text/names.html
